### PR TITLE
2496 ping pong encoding error

### LINKED ISSUE #2496: Erroneous PING and PONG encoding of enr sequence number
The besu client does not properly encode the enr sequence number in PONG responses. 

See https://github.com/hyperledger/besu/blob/e230a445bb457bbac27bf4c384da449ccbd3f043/ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/discovery/internal/PongPacketData.java#L72 

Essentially, the enr sequence is written as `writeBytes( seqAsAUint64.toBytes());`

Example Pong
```
&Pong{
            To:         Endpoint{net.ParseIP("127.0.0.1").To4(), 3322, 5544},
            ReplyTok:   common.HexToHash("0x0000000000000000000000000000000000000000000000000000000000001234").Bytes(),
            Expiration: 1136239445,
            ENRSeq:     0,
        },
```
( it's UDP 3322, TCP 5544)

Besu encodes the example above as `0xf83bcb847f000001820cfa8215a8a000000000000000000000000000000000000000000000000000000000000012348443b9a355880000000000000000`. 
```
rlpdump -hex f83bcb847f000001820cfa8215a8a000000000000000000000000000000000000000000000000000000000000012348443b9a355880000000000000000
[
  [
    7f000001,
    0cfa,
    15a8,
  ],
  0000000000000000000000000000000000000000000000000000000000001234,
  43b9a355,
  0000000000000000,
]
```

If the sequence number is `1`, it's encoded as `0xf83bcb847f000001820cfa8215a8a000000000000000000000000000000000000000000000000000000000000012348443b9a355880000000000000001`

```
 rlpdump -hex f83bcb847f000001820cfa8215a8a000000000000000000000000000000000000000000000000000000000000012348443b9a355880000000000000001
[
  [
    7f000001,
    0cfa,
    15a8,
  ],
  0000000000000000000000000000000000000000000000000000000000001234,
  43b9a355,
  0000000000000001,
]
```
So apparently the ENR sequence number is encoded as `8` bytes, not as an RLP integer. 

This was not an _obvious_ problem until geth implemented a change in how packets are decoded, in https://github.com/ethereum/go-ethereum/commit/3e6f46caec51d82aef363632517eb5842eef6db6 . Before this change: 

- Any 'junk' at the end would be ignored. Thus, the `0000000000000001` element was silently discarded, the enr sequence number was always `0`, when decoding a PONG from besu. 
- Geth would output `f3cb847f000001820cfa8215a8a000000000000000000000000000000000000000000000000000000000000012348443b9a35580` for the zero-case

```
rlpdump -hex f3cb847f000001820cfa8215a8a000000000000000000000000000000000000000000000000000000000000012348443b9a35580
[
  [
    7f000001,
    0cfa,
    15a8,
  ],
  0000000000000000000000000000000000000000000000000000000000001234,
  43b9a355,
  "",
]
```
- Geth would output the following for the `1`-case

```
rlpdump -hex f3cb847f000001820cfa8215a8a000000000000000000000000000000000000000000000000000000000000012348443b9a35501
[
  [
    7f000001,
    0cfa,
    15a8,
  ],
  0000000000000000000000000000000000000000000000000000000000001234,
  43b9a355,
  01,
]
```

Now, however, geth willl reject the 'junk' at the end. 
On output, geth willl output the following for the `0`-case
```
 rlpdump -hex f2cb847f000001820cfa8215a8a000000000000000000000000000000000000000000000000000000000000012348443b9a355
[
  [
    7f000001,
    0cfa,
    15a8,
  ],
  0000000000000000000000000000000000000000000000000000000000001234,
  43b9a355,
]
```

This issue was found in hive

Screenshot_2021-07-02 https hivetests ethdevops io

Unfortunately, the test-framework also contained another false positive, which caused the 'legit' error from not always being shown (the false positive being ` PONG 'to' endpoint mismatch: got {IP:172.17.0.7 UDP:44025 TCP:44025}, want {IP:172.17.0.7 UDP:44025 TCP:0}`). 

I think the fix would be to just use `writeLongScalar` to write the ENR sequence number. 

cc @fjl @garyschulte 

Related geth-pr with some encoding examples: https://github.com/ethereum/go-ethereum/pull/23155 
Related hive logs: https://hivetests.ethdevops.io/?page=v-pills-results-tab&suite=1625138437-05d8c3c144ebe42dbca73551d1d2f73a.json

### PR DESCRIPTION
**Pull request by jflo**
## PR description
Correct malformed ENR sequence fields in PING and PONG messages. These were encoded as byte arrays, and now are encoded as long scalar values instead.

## Fixed Issue(s)

fixes #2496 

## Changelog

- [X] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: ethereum/p2p/src/test/java/org/hyperledger/besu/ethereum/p2p/discovery/internal/PacketTest.java [hunk 1]
```diff
@@ -30,7 +30,7 @@
 public class PacketTest {
 
   private static final String VALID_PONG_PACKET =
-      "53cec0d27af44bdc0471d34c4eb631f74b502df7b5513a80a054f0d619f0417d6ba4fd4d6fb83994b95c6d0ae8b175b068a6bffc397e2b408e797069b9370ce47b153dd884b60108e686546a775ed5f85e71059a9c5791e266bd949d0dcfba380102f83bcb84b4b57a1a82040182765fa046896547d3b4259aa1a67bd26e7ec58ab4be650c5552ef0360caf9dae489d53b845b872dc8880000000000000003";
+      "3b3d56e4fcdcf714d6c29b0d521e9f4ec0dd50c73c0bbb9b44758a0a7e416ff30a3ea99de3e78a5cff82ad0a0b82030f78c6eff5a1b2a030276277b6e9b6ad7c35db1a6d5f83586a203b4537d82edc6b2820849a485e55aa06cfc680dc63c22d0002f3cb84b4b57a1a82040182765fa046896547d3b4259aa1a67bd26e7ec58ab4be650c5552ef0360caf9dae489d53b8460e5e7b603";
```
Comment (RatanRSur) [inline ethereum/p2p/src/test/java/org/hyperledger/besu/ethereum/p2p/discovery/internal/PacketTest.java:33]: Hmm, why is the whole packet different? I was expecting it to just be the last bytes.

Comment (jflo) [reply ethereum/p2p/src/test/java/org/hyperledger/besu/ethereum/p2p/discovery/internal/PacketTest.java:33]: Me too. However I learned that PacketData is RLP, and Packet is RLPx. You would be correct if that was testing PacketData - in that case you'd be able to look at it and see the difference. A Packet is encrypted and hashed.

**Review by RatanRSur — APPROVED** [review]

**Comment by matkt** [comment]
I tried on ropsten and I have this kind of issue and a lot of difficulties to find peers

org.hyperledger.besu.ethereum.rlp.MalformedRLPInputException: Invalid scalar, has leading zeros bytes (at bytes 31-40: ...60ec2e9c[88000000000000001f])
	at org.hyperledger.besu.ethereum.rlp.AbstractRLPInput.throwMalformed(AbstractRLPInput.java:194)
	at org.hyperledger.besu.ethereum.rlp.AbstractRLPInput.checkScalar(AbstractRLPInput.java:263)
	at org.hyperledger.besu.ethereum.rlp.AbstractRLPInput.checkScalar(AbstractRLPInput.java:268)
	at org.hyperledger.besu.ethereum.rlp.AbstractRLPInput.readLongScalar(AbstractRLPInput.java:290)
	at org.hyperledger.besu.ethereum.rlp.BytesValueRLPInput.readLongScalar(BytesValueRLPInput.java:23)
	at org.hyperledger.besu.ethereum.p2p.discovery.internal.PingPacketData.readFrom(PingPacketData.java:80)
	at org.hyperledger.besu.ethereum.p2p.discovery.internal.Packet.decode(Packet.java:121)
	at org.hyperledger.besu.ethereum.p2p.discovery.VertxPeerDiscoveryAgent.lambda$handlePacket$6(VertxPeerDiscoveryAgent.java:221)
	at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$2(ContextImpl.java:313)
	at io.vertx.core.impl.TaskQueue.run(TaskQueue.java:76)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)

**Review by matkt — CHANGES_REQUESTED** [review]
I tried on ropsten and I have this kind of issues and difficulties to find peers

org.hyperledger.besu.ethereum.rlp.MalformedRLPInputException: Invalid scalar, has leading zeros bytes (at bytes 31-40: ...60ec2e9c[88000000000000001f])
	at org.hyperledger.besu.ethereum.rlp.AbstractRLPInput.throwMalformed(AbstractRLPInput.java:194)
	at org.hyperledger.besu.ethereum.rlp.AbstractRLPInput.checkScalar(AbstractRLPInput.java:263)
	at org.hyperledger.besu.ethereum.rlp.AbstractRLPInput.checkScalar(AbstractRLPInput.java:268)
	at org.hyperledger.besu.ethereum.rlp.AbstractRLPInput.readLongScalar(AbstractRLPInput.java:290)
	at org.hyperledger.besu.ethereum.rlp.BytesValueRLPInput.readLongScalar(BytesValueRLPInput.java:23)
	at org.hyperledger.besu.ethereum.p2p.discovery.internal.PingPacketData.readFrom(PingPacketData.java:80)
	at org.hyperledger.besu.ethereum.p2p.discovery.internal.Packet.decode(Packet.java:121)
	at org.hyperledger.besu.ethereum.p2p.discovery.VertxPeerDiscoveryAgent.lambda$handlePacket$6(VertxPeerDiscoveryAgent.java:221)
	at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$2(ContextImpl.java:313)
	at io.vertx.core.impl.TaskQueue.run(TaskQueue.java:76)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)

**Review by matkt — APPROVED** [review]
It's fine with the last version

**Comment by jflo** [comment]
Please hold on merging. I am trying to run this with the Hive test suite to see why it fails ping related tests. There may be more fixes required.

**Comment by jflo** [comment]
This can be merged now. It improves our hive scores, but we're still not acing devp2p/discv4.

Yet.

**Pull request merged**